task1/interface_api.py

```diff
@@ -0,0 +1,65 @@
+import http
+import json
+from typing import Union, Tuple
+
+from flask import Flask, jsonify, request, Response
+
+
+class InterfaceAPI(Flask):
+    def __init__(self, import_name):
+        super(InterfaceAPI, self).__init__(import_name)
+        self.cache = {}
+        with open('resource/task-1-interfaces.json') as f:
+            data = json.loads(f.read())
+            interfaces = data['ietf-interfaces:interfaces']['interface']
+            for interface in interfaces:
+                self.cache[interface['name']] = interface
+
+
+app = InterfaceAPI(__name__)
+
+
+@app.route('/get-all-interfaces', methods=['GET'])
+def get_all_interfaces() -> Response:
+    all_interfaces = app.cache.values()
+    return jsonify(list(all_interfaces))
+
+
+@app.route('/get-interface/<path:interface_name>', methods=['GET'])
+def get_some_interface(interface_name: str) -> Union[Response, Tuple[str, int]]:
+    interface = app.cache.get(interface_name)
+    if interface:
+        return jsonify(interface)
+    else:
+        return f'Interface {interface_name} does not exist.', http.HTTPStatus.NOT_FOUND
+
+
+@app.route('/get-interfaces', methods=['POST'])
+def post_several_interfaces() -> Response:
+    all_interfaces = app.cache.values()
+    req_data = request.get_json()
+    result = []
+    for interface in all_interfaces:
+        for input_interface in req_data['input']['interfaces']:
```

Comment:
@patrikprovaznik invert this cycle and if we have name of input request then get it from your cache.. which is much faster if we don t need to iterate through all interfaces.